src/main/java/org/apache/sysds/runtime/instructions/fed/FEDInstructionUtils.java

```diff
@@ -232,12 +232,7 @@ else if((tinst.input1.isMatrix() && ec.getCacheableData(tinst.input1).isFederate
 			}
 			else if(inst instanceof VariableCPInstruction ){
 				VariableCPInstruction ins = (VariableCPInstruction) inst;
-				if(ins.getVariableOpcode() == VariableOperationCode.Write
-					&& ins.getInput1().isMatrix()
-					&& ins.getInput3().getName().contains("federated")){
-					fedinst = VariableFEDInstruction.parseInstruction(ins);
-				}
-				else if(ins.getVariableOpcode() == VariableOperationCode.CastAsFrameVariable
+				if(ins.getVariableOpcode() == VariableOperationCode.CastAsFrameVariable
```

Comment:
The federated writing happens inside the `CacheableData<T>.exportData()` method, therefore we don't need to replace the `VariableCPInstruction`.